feature-libs/asm/components/asm-bind-cart/asm-bind-cart.component.ts

```diff
@@ -86,13 +56,14 @@ export class AsmBindCartComponent implements OnInit, OnDestroy {
    * Bind the input cart number to the customer
    */
   bindCartToCustomer() {
-    const subscription = this.loading$
-      .asObservable()
+    const subscription = of(this.loading$.getValue())
       .pipe(
-        take(1),
         filter((loading) => !loading && this.cartId.valid),
         tap(() => this.loading$.next(true)),
-        concatMap(() => this.bindCartToCurrentUser$.execute(this.cartId.value)),
+        concatMap(() => this.asmBindCartFacade.bindCart(this.cartId.value)),
+        tap(() => {
+          this.multiCartFacade.reloadCart(OCC_CART_ID_CURRENT);
```

Comment:
When we all reloadCart in the tap(), are we sure at that point that the cart binding was a success?

Otherwise, it could also improve readability to group together the logic to be applied when the cart binding is a success.  We'd just need to move the reloadCart before we display the success message display.  There might also be something I missed.  Perhaps there is a reason for how it is implemented like error catching.

Reply:
Yes, the binding must be a success.  The `tap()`'s "next" listener executes only when  `asmBindCartFacade.bindCart()` emits.  If it "error"s, then the tap is passed by.  If the filter above causes a "complete", the tap would also be passed by.

I agree it would be more readable to move the `tap()`'s content to the `subscribe()`'s "next" listener.